src/components/ecl-button/ecl-button.tsx

```diff
@@ -42,14 +47,27 @@ export class EclButton {
         this.el.querySelector('.ecl-icon').classList.add(`ecl-button__${slot.substring(0, 5) + '-' + slot.substring(5)}`, `sc-ecl-button-${this.theme}`);
       }
     }
+
+    const iconBefore = this.el.querySelector('[slot="icon-before"]');
+    const iconAfter = this.el.querySelector('[slot="icon-after"]');
+
+    this.hasIconBefore = !!iconBefore;
+    this.hasIconAfter = !!iconAfter;
+
+    // Check for label content
+    const labelEl = this.el.querySelector('.ecl-button__label');
+    if (labelEl) {
+      this.hasLabelContent = labelEl.textContent.trim() !== '';
```

Comment:
with this we are going to add this class everwhere we are going to have a label, but we only need it when we have an icon instead, can we limit then its usage to the use cases where we really need it not being an ecl selector..?

Reply:
I've just noticed something, we are missing the ecl-button--icon-only class when the hideLabel is true, therefore you can use this instead of this "alien" selector that you introduced, as a :not(.ecl-button--icon-only). Please check that once you will introduce this selector the icon will be 1.5rem and that it will be centered, currently it's not.

Reply:
and yes, it has now that size increased, made sure it works properly